### DISCUSSION TITLE
How to install a plugin from a local folder?

### DISCUSSION THREAD

**Discussion by bartenra** [dotansimha/graphql-code-generator#5779] [Q&A] [answered]
Hello there!

I wrote a simple plugin to generate (Python) types for a GraphQL server called [Ariadne](https://ariadnegraphql.org/). Before making it public I'd like to publish it as an independent plugin and iterate on the implementation.

When I use `npm pack` and then `npm install` the resulting `.tgz` file, I can use it with GraphQL Code Generator.

However, when I use `npm install <folder>` I get an error message:

```
 Error: Cannot use GraphQLEnumType "Lang" from another module or realm.

    Ensure that there is only one instance of "graphql" in the node_modules
    directory. If different versions of "graphql" are the dependencies of other
    relied on modules, use "resolutions" to ensure only one version is installed.
```

Does anyone have a setup with a plugin installed from a local folder that uses `npm` to install that local folder inside another project?

**Comment by bartenra** [comment]
The strange thing is that it is a `yarn` error while I am using `npm`.

**Comment by tvvignesh** [comment, marked as answer]
@bartenra Hi 👋🏼 I had a similar issue sometime back which I had raised to @dotansimha and ofcourse, the suggestion was to keep using the tgz file. Attaching the screenshot for your reference

![image](https://user-images.githubusercontent.com/1165845/120148984-aba14f80-c206-11eb-9b9f-33ee6ebf544d.png)

Currently, this is the only workaround we see unfortunately.
